crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -515,16 +544,79 @@ fn create_get_pending_operations_message() -> Result<Message, ConversionError> {
-    let ops = Operations::try_new(ops_dir, C8Y_CLOUD)?;
+    let dir_split: Vec<&str> = cfg_dir
+        .to_str()
+        .ok_or_else(|| ConversionError::DirPathComponentError {
+            dir: cfg_dir.to_path_buf(),
+        })?
+        .split('/')
+        .collect();
+
+    if dir_split.len() == 6 {
+        let stopic = format!("{SMARTREST_PUBLISH_TOPIC}/{}", dir_split[5]);
+
+        create_ops_message(
+            Operations::try_new(cfg_dir, None)?,
+            Topic::new_unchecked(&stopic),
+        )
+    } else {
+        let ops_dir = format!("{}/{SUPPORTED_OPERATIONS_DIRECTORY}", cfg_dir.display());
+        create_ops_message(
+            Operations::try_new(ops_dir, Some(C8Y_CLOUD))?,
+            Topic::new_unchecked(SMARTREST_PUBLISH_TOPIC),
+        )
+    }
+}
+
+fn create_ops_message(ops: Operations, topic: Topic) -> Result<Message, ConversionError> {
     let ops = ops.get_operations_list();
     let ops = ops.iter().map(|op| op as &str).collect::<Vec<&str>>();
-
     let ops_msg = SmartRestSetSupportedOperations::new(&ops);
-    let topic = Topic::new_unchecked(SMARTREST_PUBLISH_TOPIC);
+
     Ok(Message::new(&topic, ops_msg.to_smartrest()?))
 }
 
+fn create_supported_child_operations_fragments_message(
+    children: &mut HashSet<String>,
+    cfg_dir: &Path,
+) -> Result<Vec<Message>, ConversionError> {
+    let mut mqtt_messages = Vec::new();
+    let ops_dir = format!("{}/{SUPPORTED_OPERATIONS_DIRECTORY}", cfg_dir.display());
+    let path: PathBuf = ops_dir.into();
+    let child_entries = fs::read_dir(&path.join(C8Y_CLOUD))
+        .map_err(|_| ConversionError::ReadDirError {
+            dir: PathBuf::from(&path),
+        })?
+        .map(|entry| entry.map(|e| e.path()))
+        .collect::<Result<Vec<PathBuf>, _>>()?
+        .into_iter()
+        .filter(|path| path.is_dir())
+        .collect::<Vec<PathBuf>>();
+
+    for cdir in child_entries {
+        let ops = Operations::try_new(cdir.clone(), None)?;
+        let ops = ops.get_operations_list();
+        let ops = ops.iter().map(|op| op as &str).collect::<Vec<&str>>();
+        let ops_msg = SmartRestSetSupportedOperations::new(&ops);
+        if let Some(id) = cdir.iter().next_back() {
+            if id.to_str().is_some() {
+                let child_id = id.to_str().unwrap();
+                if !children.contains(child_id) {
+                    children.insert(child_id.to_string());
+                    mqtt_messages.push(Message::new(
+                        &Topic::new_unchecked(SMARTREST_PUBLISH_TOPIC),
+                        format!("101,{child_id},{child_id},thin-edge.io-child"),
+                    ));
+                }
+                let topic_str = format!("{SMARTREST_PUBLISH_TOPIC}/{}", child_id);
+                let topic = Topic::new_unchecked(&topic_str);
+                mqtt_messages.push(Message::new(&topic, ops_msg.to_smartrest()?));
+            }
+        }
```

Comment:
I would expect a function that produces the message for a single child.

When an operation is added to a child, do we send as many `101` messages as child devices?

Reply:
> I would expect a function that produces the message for a single child.
I added a function to create the `message` for a single child.
> 
> When an operation is added to a child, do we send as many `101` messages as child devices?
No, `101` will be sent on adding or removing an operation.